python/cuml/test/conftest.py

```diff
@@ -108,37 +110,45 @@ def fail_on_bad_cuml_array_name(monkeypatch, request):
 
     def patched__setattr__(self, name, value):
 
-        supported_type = get_supported_input_type(value)
-
-        if name == 'idf_':
-            # We skip this test because idf_' for tfidf setter returns
-            # a sparse diagonal matrix and getter gets a cupy array
-            # see discussion at:
-            # https://github.com/rapidsai/cuml/pull/2698/files#r471865982
-            pass
-        elif (supported_type == CumlArray):
-            assert name.startswith("_"), "Invalid CumlArray Use! CumlArray \
-                attributes need a leading underscore. Attribute: '{}' In: {}" \
-                    .format(name, self.__repr__())
-        elif (supported_type == cp.ndarray and
-              cupyx.scipy.sparse.issparse(value)):
-            # Leave sparse matrices alone for now.
+        if name == 'classes_' and isinstance(self,
```

Comment:
Is there a way to solve this locally in LabelEncoder instead? What about storing classes as some other name like double-underscore __classes and then having a property like:
```
@property
def classes(self):
 return self.__classes
```

This test code seems to be saying that classes_ doesn't fit the CumlArray pattern (true!) so I worry if there are possible issues elsewhere without a fixup?

Reply:
The test was failing as a result of changing the LabelEncoder to import from `Base` instead of `object` 

Since `self.classes_` contains strings and is expected to be CumlArray , it results in an error. Making the change in LabeEncoder might break other things that depend on `self.classes_` to be present. Discussed this change with @cjnolet and @dantegd too.